packages/near-membrane-base/src/blue.ts

```diff
@@ -34,56 +35,42 @@ import {
     RedConstructor,
     RedFunction,
     RedValue,
+    BlueArrayOrObject,
+    RedArrayOrObject,
 } from './types';
 
-function renameFunction(provider: RedFunction, receiver: BlueFunction) {
-    try {
-        // a revoked proxy will break the membrane when reading the function name
-        const nameDescriptor = ReflectGetOwnPropertyDescriptor(provider, 'name')!;
-        ReflectDefineProperty(receiver, 'name', nameDescriptor);
-    } catch {
-        // intentionally swallowing the error because this method is just extracting the
-        // function in a way that it should always succeed except for the cases in which
-        // the provider is a proxy that is either revoked or has some logic to prevent
-        // reading the name property descriptor.
-    }
-}
-
 const ProxyCtor = Proxy;
 const { isArray: isArrayOrNotOrThrowForRevoked } = Array;
 
-function createBlueShadowTarget(target: BlueProxyTarget): BlueShadowTarget {
+// eslint-disable-next-line no-shadow
+enum TargetTraits {
```

Comment:
typescript for bitwise is a little funky at a very least, but the big issue here is that typescript uses the type declarations as runtime values (when custom values are provided - like the following). What this means is that because we do toString() on the red function, if the types are imported, the bitwise are lost in the sea. As a result, the bitwise types must be declared in the scope (which means duplication), but that's not a very big deal, it is just a small list.

Reply:
Add: `None = 0,` at the beginning

Reply:
That's super interesting that the Type becomes runtime values.